### PR TITLE
Fix macro nested repetitions

### PR DESCRIPTION
**Pull request by powerboat9**
Depends on #2260

### CODE REVIEW

**Comment by powerboat9** [comment]
It looks like there's still an issue with metavariables in nested repetitions where multiple repetitions are matched 0 times.

**Comment by powerboat9** [comment]
Created #2268, since the described issue existed before #2260

**Comment by philberty** [comment]
There are some conflicts now @powerboat9

**Comment by powerboat9** [comment]
Conflict resolved

File: gcc/rust/expand/rust-macro-expand.cc [hunk 1]
```diff
@@ -617,7 +614,7 @@ MacroExpander::match_n_matches (Parser<MacroInvocLexer> &parser,
 		// `insert_fragment` with the difference that we are not
 		// creating a metavariable, but a repetition of one, which is
 		// really different.
```
[the remark this answers is not in the document]
Comment (CohenArthur) [reply gcc/rust/expand/rust-macro-expand.cc:616]: Apart from this I'm in favor of merging this PR :)

**Review by P-E-P — APPROVED** [review]

**Comment by powerboat9** [comment]
@CohenArthur had to fix merge conflicts

**Comment by powerboat9** [comment]
Oops

**Pull request merged**